MN/Algoritmi/Metode-Iterative/Gauss-Seidel/GaussSeidel.m

```diff
@@ -0,0 +1,49 @@
+function [x step] = GaussSeidel(A, b, x0, tol, max_iter)
+    % use this algorithm only if it converges
+    
+    % sanity checks
+    N = tril(A);
+    P = N - A;
+    Ggs = inv(N) * P;
+    % calculate the spectral radius of Ggs
+    sr = max(abs(eig(Ggs)));
+    % check if the algorithm converges
+    
+    if  (sr - 1) >= eps
+        x = NaN;
+        step = -1;
+        disp('Matricea nu converge');
+        return;
+    endif
+    
+    % for the function to work make sure
+    % that b and x0 are column vectors
+    
+    n = length(b);
+    x = x0;
+    % iterate to the maximum number of iterations 
+    for step = 1 : max_iter
+        % iterate through every x(i)
+        for i = 1 : n
+            % new_values_sum represents the first sum
+            % of the Gauss-Seidel algorithm
+            new_values_sum = A(i, 1 : i - 1) * x(1 : i - 1);
+            
+            % old_values_sum represents the second sum
+            % of the Gauss-Seidel algorithm
+            old_values_sum = A(i, i + 1 : n) * x0(i + 1 : n);
+            
+            x(i) = (b(i) - (old_values_sum + new_values_sum)) / A(i, i);
+        endfor
+        % when the new values get close 
```

Comment:
linie verticala de 80 sau 100 de caractere. cred ca astea 4 comentarii se pot scrie in 2 randuri.